go/libkb/user.go

```diff
@@ -941,7 +941,10 @@ func CanLogout(mctx MetaContext) (res keybase1.CanLogoutRes) {
 		return res
 	}
 
-	if err := CheckCurrentUIDDeviceID(mctx); err != nil {
+	// since this is on the critical path to logout,
+	// use a short timeout in case we're offline.
+	m, _ := mctx.WithTimeout(600 * time.Millisecond)
+	if err := CheckCurrentUIDDeviceID(m); err != nil {
```

Comment:
I'm worried that if we are online, but 600ms is not enough for the API call to finish, we might not be able to success early on `case DeviceNotFoundError, ...:` and keep failing on `LoadHasRandomPw` if user really doesn't have a device or doesn't exist anymore. So on slow connection you will be in permanent state of timeouting on `CheckCurrentUIDDeviceID` and then failing on `LoadHasRandomPw`.

`libkb/api.go` already uses `ConnectivityMonitor.IsConnected` to reduce timeout to 5 seconds if we are not connected to gregor (so when we are potentially offline). Maybe 5 seconds is not bad here? Especially with the GUI spinner.

Reply:
Sorry I didn't bring this up yesterday, I only realized this now. Let's also see what Patrick thinks.

Reply:
I agree I don't think this is the right startegy. 600msec is not enough. I think a spinner is the right strategy.